flow360/component/geometry_metadata.py

```diff
@@ -0,0 +1,141 @@
+"""Geometry metadata pipeline output models"""
+
+from collections import defaultdict
+from typing import DefaultDict, List, Literal, Optional, Set, Union
+
+import pydantic as pd
+
+from flow360.component.simulation.framework.entity_registry import EntityRegistry
+from flow360.component.simulation.primitives import Edge, Surface
+from flow360.component.simulation.web.asset_metadata import MetadataBaseModel
+from flow360.log import log
+
+
+class _KeyValuePair(MetadataBaseModel):
+    key: str = pd.Field()
+    value: str = pd.Field()
+
+
+class _FaceEdgeMeta(MetadataBaseModel):
+    id: str = pd.Field()
+    attributes: Optional[List[_KeyValuePair]] = pd.Field(None)
+
+
+class _FaceInfo(MetadataBaseModel):
+    type: Literal["face"] = pd.Field("face", frozen=True)
+    metadata: _FaceEdgeMeta = pd.Field()
+
+
+class _EdgeInfo(MetadataBaseModel):
+    type: Literal["edge"] = pd.Field("edge", frozen=True)
+    metadata: _FaceEdgeMeta = pd.Field()
+
+
+class _BodyMeta(MetadataBaseModel):
+    faces: List[_FaceInfo] = pd.Field()
+    edges: List[_EdgeInfo] = pd.Field()
+
+
+class _BodyInfo(MetadataBaseModel):
+    type: Literal["body"] = pd.Field("body", frozen=True)
+    metadata: _BodyMeta = pd.Field()
+
+
+class _GeometryMetadataModel(MetadataBaseModel):
+    """
+    A model to handle geometry metadata including bodies and processing of entity types like faces or edges.
+    """
+
+    bodies: list[_BodyInfo] = pd.Field()
+    version: str = pd.Field()
+
+    def process_metadata_for_given_type(self, entity_type_name: Literal["faces", "edges"]):
+        """
+        Process the metadata for the given entity type (faces or edges) and return grouped information.
+
+        Args:
+            entity_type_name (Literal["faces", "edges"]): The type of entity to process ('faces' or 'edges').
+
+        Returns:
+            DefaultDict[str, DefaultDict[str, Set[str]]]: Grouped information based on entity attributes.
+
+        Metadata JSON example:
+            {
+                "faceName": {
+                    "MyFace1": {"body01_face001"},
+                    "MyFace2": {"body01_face002"},
+                },
+                "boundaryName": {
+                    "Wing1": ["body01_face002"],
+                    "Wing2": ["body01_face003"],
+                },
+                "__all__": {"body01_face001", "body01_face002", "body01_face003"},
+            }
```

Comment:
can we stick to one docstring format? I think this will be important for autodocs generation. We use numpy docstring style. Not very important for this PR, but something to keep track

Reply:
Yes. I just explored some documentation last week and yes we should use numpy style. Most docstring currently is just for explaining, not serious.